### ISSUE TITLE
Some bugs and differences with the original release

### ISSUE THREAD

**Issue by LeonarthCG** [evanbowman/Hero-port-gba#1]
Just finished playing through this port and, for the most part, it is pretty faithful, at least as far as I could tell, having played Hero Core to completion plenty of times. Most of the inconsistencies I noticed were listed on the github page, but some were not, so I figured I would mention them.
- The boss Zero is not present.
- The Eliminator can spawn even if you are at level 0. It is not meant to spawn until you hit level 1 (and also it can't run away if you are level 9).
- The shapeshift ability seems to be missing from the game.
- Flip Hero can't fit through the center columns of ​​the generators. I understand that the hitboxes have been modified, but being unable to fit through these tiles does affect at least one room transition that is no longer possible.
- The Pattern scanner and Ship scan are missing.
- The terminals that display text are also missing, but I guess that's to be expected, since the tutorial text and such is mentioned to be missing in the feature list.
- The room transition tiles for save rooms are the same as for any other room, unlike in the original game.

I also ran into a few bugs:
- The Star Splitter boss keeps respawning, dropping further Suit upgrades. The door for the boss room does not show up again, though, so the player can just leave.
- The generator for the core barriers respawns, making it impossible to reach the final boss without abusing iframes, as the barriers never go away.
- By transitioning screens around the space outside of the station (by exiting through the top left) it is very easy to end up inside a wall. Being in this area also makes the character indicator on the minimap get stuck on the very top left of the screen.
- The player can pause while dying. If they do this and then unpause, the death animation starts all over again.
- I think I was able to use the blade while heat was not 0, before having suit level 2. Perhaps I'm remembering wrong for this one, though.

Anyway, as I said, I think the port is shaping up very well so far. Are there plans to continue the development now that the jam is over?

**Comment by evanbowman** [comment]
Thank you for the feedback! Yeah I intend to keep working on it! But I'm going to take a break for a couple months as I've been a bit overworked haha. I already wrote more than half of the code for the enemy patterns for hard mode, so I intend to at least finish hard mode and add the text and cutscenes. Then boss rush mode would be easy enough to add. I hadn't planned on forever mode and the GBA probably couldn't handle the ReallyJoelsDad difficulty due to sprite limits :)
Thanks for playing and I'll try to address all of these issues soon, particularly the bugs. For the issues involving incompleteness, I may create separate issues for each one so that I can track them. The star splitter mistake is particularly nasty and I'll have to fix that.
I never ran into the generator for the core barriers respawning, I'll look into that one too!

**Comment by evanbowman** [comment]
Do you happen to have a .sav file from when the generator near the core respawns?

**Comment by evanbowman** [comment]
If both the StarSplitter and some generators are respawning, and they didn't during my testing, I think that array that I'm using to store persistent changes to overworld objects may simply be full and I can bump the size, but if you had a .sav file from a playthrough that would help me confirm the cause.
~~https://github.com/evanbowman/Hero-port-gba/blob/master/source/engine.hpp#L233~~
Edit: permalink: https://github.com/evanbowman/Hero-port-gba/blob/03fb85dfe220c4b47581ec29c0e795b661c02715/source/engine.hpp#L233

**Comment by evanbowman** [comment]
The code tries to remove the star splitter and generators persistently, https://github.com/evanbowman/Hero-port-gba/blob/master/source/objects/enemies/boss/starSplitter.cpp#L463
So it's almost certainly an issue with an insufficiently sized buffer for storing objects removed from the map. I'll just bump the size!

**Comment by evanbowman** [comment]
I think I may have fixed the issue with the star splitter and generator in the latest release. Not backwards compatible and won't load existing saves, though. I'll address the other issues in the coming weeks.

**Comment by LeonarthCG** [comment]
Sadly I don't have the save anymore, since I started a new game right away after beating it so I could check if Zero would spawn. I do know that the version I am playing is whatever is up for download on itch, since I just went and redownloaded it to check if the files were identical.

If it helps, I did beat Star Splitter a few extra times before I noticed the generator issue, and my suit level was up to 4 or 5, displaying some random tile instead of numbers. This also did happen after I had opened every door, destroyed every (other) generator, killed every boss (other than the offending one) and collected everything there was to collect except for the Eliminator's level up, so if your buffer's size would be at risk it would be after the player has done that much exploring, I assume.

Either way, I will try to replay the game again eventually, so I'll let you know if I spot anything.

**Comment by evanbowman** [comment]
Thanks! I reproduced the issue and confirmed the cause this morning. The latest patch on this repo fixes the issue.

**Comment by evanbowman** [comment]
I'll leave this issue open until I can look into the other things.

**Comment by zlago** [comment]
(is this the right issue for a bug report? i dont wanna open another issue..)
i died right after defeating tetron, and i couldnt move after i respawned, softlocking the game (this was on the homebrew hub version)

**Comment by evanbowman** [comment]
> (is this the right issue for a bug report? i dont wanna open another issue..) i died right after defeating tetron, and i couldnt move after i respawned, softlocking the game (this was on the homebrew hub version)

A new issue would be preferable in the future, but it's fine this time. This bug will be fun to reproduce, so you destroyed tetron, then got hit by a laser, in that order?

**Comment by zlago** [comment]
> This bug will be fun to reproduce, so you destroyed tetron, then got hit by a laser, in that order?

yeah, exactly like that, except instead of a laser it was one of those bullets that burst out of tetron when he lowers his gun.
wouldnt the fix be as simple as making sure all bullets are destroyed before player controls are locked though?

**Comment by evanbowman** [comment]
Yeah, that'd be the obvious way to fix things simply, but I need to check the original game to make sure that the pc version also clears bullets when tetron dies, as I'm trying to create an accurate port
